fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -2309,6 +2354,68 @@ void MultiVersionDatabase::DatabaseState::close() {
 	});
 }
 
+namespace {
+
+const char* inializationStateToString(MultiVersionDatabase::InitializationState initState) {
+	switch (initState) {
+	case MultiVersionDatabase::InitializationState::INITIALIZING:
+		return "initializing";
+	case MultiVersionDatabase::InitializationState::INITIALIZATION_FAILED:
+		return "initialization_failed";
+	case MultiVersionDatabase::InitializationState::CREATED:
+		return "created";
+	case MultiVersionDatabase::InitializationState::INCOMPATIBLE:
+		return "incompatible";
+	case MultiVersionDatabase::InitializationState::CLOSED:
+		return "closed";
+	default:
+		ASSERT(false);
+		return "invalid_state";
+	}
+}
+
+} // namespace
+
+Standalone<StringRef> MultiVersionDatabase::DatabaseState::getClientStatus(
+    ErrorOr<Standalone<StringRef>> dbContextStatus) {
+	json_spirit::mObject statusObj;
+	statusObj["InitializationState"] = inializationStateToString(initializationState);
+	if (initializationState == InitializationState::INITIALIZATION_FAILED) {
+		statusObj["InitializationError"] = initializationError.code();
+	}
+	json_spirit::mArray clientArr;
+	for (auto [protocolVersion, client] : this->clients) {
+		json_spirit::mObject clientDesc;
+		clientDesc["ProtocolVersion"] = format("%llx", client->protocolVersion.version());
+		clientDesc["ReleaseVersion"] = client->releaseVersion;
+		clientDesc["ThreadIndex"] = client->threadIndex;
+		clientArr.push_back(clientDesc);
+	}
+	statusObj["AvailableClients"] = clientArr;
+	statusObj["ConnectionRecord"] = connectionRecord.toString();
+	statusObj["ClusterId"] = clusterId;
+	if (dbProtocolVersion.present()) {
+		statusObj["ProtocolVersion"] = format("%llx", dbProtocolVersion.get().version());
+	}
+	bool dbContextHealthy = false;
+	if (initializationState != InitializationState::INITIALIZATION_FAILED) {
+		if (dbContextStatus.isError()) {
+			statusObj["ErrorRetrievingDatabaseStatus"] = dbContextStatus.getError().code();
```

Comment:
`name()` would be more convenient than `code()`, as

Reply:
It would be more human readable, but not sure which one is more convenient for processing in code if necessary. We give an API fdb_get_error for translation to name, but not the other way around.

Reply:
Maybe I'm near-sighted but I don't foresee any situation where we need to translate an error name *back to* error code.

Reply:
XP & GS work with FDB error codes, if they want to know what error it is they compare against codes. In any case, I will ask what they prefer. We can can also report both